Fix unit tests for existing functionality (fixes invalid platform) & add new 'setAttributes' method for modifying existing user attributes.

race opened the pull request:
Updates the unit tests to fix the invalid platform for the current version (1.x).

Moved the region to a new SNS_REGION variable for easier modification during local testing.

Added a new 'setAttributes' method to allow updating an existing user via their EndpointArn (useful for refreshing token or user information) with a new test case and two new events (attributesUpdated and AttributesUpdateFailed).

evanshortiss commented on lib/interface.js:
```diff
@@ -146,6 +148,35 @@ Interface.prototype.getUser = function(endpointArn, callback) {
+  var self = this;
+  if (typeof attributes == 'object') {
+    if (typeof attributes.CustomUserData == 'object') {
+      attributes.CustomUserData = JSON.stringify(attributes.CustomUserData);
```
Could this be wrapped in a try/catch to ensure it's safe?

race replied:
Of course - I'd be happy to :-) will be able to push the update later this morning.

evanshortiss commented:
Great work @race! Just one edit suggested and I'm happy to merge.

race commented:
I've added a try/catch to ensure that parsing the JSON is safe and also implemented additional error handling should someone accidentally supply an invalid type for the second argument of the setAttributes method.

evanshortiss commented:
Thanks @race 

I'll publish a new version to NPM soon so it's readily available for easy install.

race commented:
Thank you!

evanshortiss commented:
@race Published as 1.1.0

race commented:
Awesome, thanks again and appreciate the publish notice.
